lib/spack/spack/solver/asp.py

```diff
@@ -2169,14 +2161,16 @@ def reorder_flags(self):
             sorted_sources = sorted(sources, key=lambda s: order.index(s))
 
             # add flags from each source, lowest to highest precedence
-            flags = collections.defaultdict(lambda: [])
+            flags = []
```

Comment:
Should there be an explanation for why this isn't a dictionary?

Reply:
@tldahlgren I'm not sure why. What part needs to be explained? We're just checking the flag values, so which grab the entire dictionary?

Reply:
Resolving since we discussed this offline too.